src/image_cache/mod.rs

```diff
@@ -42,19 +42,20 @@ pub mod errors {
 pub use self::errors::Result;
 use self::errors::*;
 
-pub fn get_image_size_estimate(dimensions: (u32, u32)) -> u32 {
+pub fn get_image_size_estimate(width: u32, height: u32) -> isize {
 	// counting all the mipmaps would add an additionnal multiplier of around ~1.6
 	// but only the gpu textures have mip maps so just multiply by 1.5
-	((dimensions.0 * dimensions.1 * 4) as f32 * 1.5) as u32
+	((width * height * 4) as f32 * 1.5) as isize
```

Comment:
Yep this is nice.
Two things here:
1, Why is the size an `isize` instead of a `usize`?
2, The intermediate value might as well be an `f64` to minimize info loss.

Reply:
I changed it to an `isize` because it was always converted to an `isize` afterwards (to allow subtracting with underflow I believe).

`f32` has 6–9 significant decimal digits. I thinkt that should be enough for an _estimate_ of the image size. To get more precision, we could return `width * width * 6` :)

Reply:
The maximum jpeg size of 2^32 pixels would produce a size estimate of 2^32 * 6 which cannot be represented by a 32bit value.

So I do like the idea to multiply by 6, but as it reduces the readability the comment above should be extended to describe that the 6 is for `channel_count * mipmap_factor` which is `4 * 1.5`. Or you could introduce local `const`s. Whichever you prefer.